src/fork_gc.c

```diff
@@ -1271,45 +1234,16 @@ static int periodicCb(RedisModuleCtx *ctx, void *privdata) {
       gcrv = 1;
     }
     close(gc->pipefd[GC_READERFD]);
-    if (FGC_haveRedisFork()) {
-
-      if (gc->type == FGC_TYPE_NOKEYSPACE) {
-        // If we are not in key space we still need to acquire the GIL to use the fork api
-        RedisModule_ThreadSafeContextLock(ctx);
-      }
-
-      if (!FGC_lock(gc, ctx)) {
-        if (gc->type == FGC_TYPE_NOKEYSPACE) {
-          RedisModule_ThreadSafeContextUnlock(ctx);
-        }
-
-        return 0;
-      }
-
-      // KillForkChild must be called when holding the GIL
-      // otherwise it might cause a pipe leak and eventually run
-      // out of file descriptor
-      RedisModule_KillForkChild(cpid);
-
-      if (gc->type == FGC_TYPE_NOKEYSPACE) {
-        RedisModule_ThreadSafeContextUnlock(ctx);
-      }
-
-      FGC_unlock(gc, ctx);
-
-    } else {
-      pid_t id = wait4(cpid, NULL, 0, NULL);
-      if (id == -1) {
-        RedisModule_Log(ctx, "warning", "an error occurred when waiting for fork GC to terminate,"
-                        " pid:%d", cpid);
-      }
-    }
+    // KillForkChild must be called when holding the GIL
+    // otherwise it might cause a pipe leak and eventually run
+    // out of file descriptor
+    RedisModule_ThreadSafeContextLock(ctx);
+    RedisModule_KillForkChild(cpid);
+    RedisModule_ThreadSafeContextUnlock(ctx);
```

Comment:
We need to follow the existing locking logic on this version (type check + using `FGC_lock`).
On 2.6 we didn't have the spec r/w lock.